File: app/src/main/java/com/jiangkang/ktools/web/WebActivity.java
package com.jiangkang.ktools.web;

import android.annotation.TargetApi;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ImageView;
import android.widget.TextView;

import com.jiangkang.ktools.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

import static android.webkit.WebSettings.MIXED_CONTENT_ALWAYS_ALLOW;

public class WebActivity extends AppCompatActivity implements WebContract.IView{


    private static final String TAG = "WebActivity";
    @BindView(R.id.iv_title_left)
    ImageView ivTitleLeft;
    @BindView(R.id.tv_title_middle)
    TextView tvTitleMiddle;
    @BindView(R.id.tv_title_right)
    TextView tvTitleRight;
    @BindView(R.id.web_container)
    WebView webContainer;
    private String launchUrl;


    /*
    * launchUrl : 网址
    *
    * */
    public static void launch(Context context, Bundle bundle) {
        Intent intent = new Intent(context, WebActivity.class);
        if (bundle != null) {
            intent.putExtras(bundle);
        }
        context.startActivity(intent);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web);
        ButterKnife.bind(this);
        initVar();

    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    private void initVar() {
        launchUrl = getIntent().getStringExtra("launchUrl");

        Log.d(TAG, "initVar: launchUrl = " + launchUrl);

        webContainer.getSettings().setMixedContentMode(MIXED_CONTENT_ALWAYS_ALLOW);
        webContainer.setWebChromeClient(new KWebChromeClient(this));
        webContainer.setWebViewClient(new KWebViewClient(this));

        //webContainer.setLongClickable(false);
        //webContainer.setOnLongClickListener(new View.OnLongClickListener() {
        //  @Override public boolean onLongClick(View v) {
        //    return true;
        //  }
        //});

        WebView.setWebContentsDebuggingEnabled(true);

        webContainer.getSettings().setJavaScriptEnabled(true);
        webContainer.getSettings().getAllowFileAccessFromFileURLs();

        webContainer.addJavascriptInterface(new KJavaInterface(this),"jk");

        webContainer.loadUrl(launchUrl);

    }


    @OnClick(R.id.iv_title_left)
    public void onIvTitleLeftClicked() {
        if (webContainer.canGoBack()){
            webContainer.goBack();
        }else {
            finish();
        }
    }

    @OnClick(R.id.tv_title_right)
    public void onTvTitleRightClicked() {

    }

    @Override
    public void onBackPressed() {
        if (webContainer.canGoBack()) {
            webContainer.getSettings().setCacheMode(WebSettings.LOAD_CACHE_ELSE_NETWORK);
            webContainer.goBack();
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public TextView getTvTitle() {
        return tvTitleMiddle;
    }

    @Override
    public ImageView getIvBack() {
        return ivTitleLeft;
    }


}

File: app/src/main/java/com/jiangkang/ktools/web/WebActivity.kt
package com.jiangkang.ktools.web

import android.annotation.TargetApi
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.webkit.WebSettings
import android.webkit.WebSettings.MIXED_CONTENT_ALWAYS_ALLOW
import android.webkit.WebView
import android.widget.ImageView
import android.widget.TextView
import butterknife.ButterKnife
import butterknife.OnClick
import com.jiangkang.ktools.R
import kotlinx.android.synthetic.main.activity_web.*

class WebActivity : AppCompatActivity(), WebContract.IView {

    //网址
    private var launchUrl: String? = null


    private var mContext = this

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web)
        ButterKnife.bind(this)
        initVar()
    }


    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    private fun initVar() {
        launchUrl = intent.getStringExtra("launchUrl")

        Log.d(TAG, "initVar: launchUrl = " + launchUrl!!)

        webContainer?.apply {
            webChromeClient = KWebChromeClient(mContext)
            webViewClient = KWebViewClient(mContext)
            addJavascriptInterface(KJavaInterface(mContext), "jk")
        }

        webContainer?.settings?.apply {
            mixedContentMode = MIXED_CONTENT_ALWAYS_ALLOW
            javaScriptEnabled = true
            allowFileAccessFromFileURLs = true
            setGeolocationEnabled(true)
            allowFileAccess = true
            allowFileAccessFromFileURLs = true
        }


        WebView.setWebContentsDebuggingEnabled(true)

        webContainer?.loadUrl(launchUrl)

    }


    @OnClick(R.id.iv_title_left)
    fun onIvTitleLeftClicked() {
        if (webContainer!!.canGoBack()) {
            webContainer!!.goBack()
        } else {
            finish()
        }
    }

    @OnClick(R.id.tv_title_right)
    fun onTvTitleRightClicked(): TextView {
        return tv_title_right
    }

    override fun onBackPressed() {
        if (webContainer!!.canGoBack()) {
            webContainer!!.settings.cacheMode = WebSettings.LOAD_CACHE_ELSE_NETWORK
            webContainer!!.goBack()
        } else {
            super.onBackPressed()
        }
    }

    override fun getTvTitle(): TextView? {
        return tv_title_middle
    }

    override fun getIvBack(): ImageView? {
        return iv_title_left
    }


    override fun onDestroy() {
        super.onDestroy()
    }

    companion object {

        private val TAG = "WebActivity"

        /**
         *
         * @param bundle
         * launchUrl : 网址
         * imgClickable : 点击图片是否显示大图
         *
         * @return
         */
        fun launch(context: Context, bundle: Bundle?) {
            val intent = Intent(context, WebActivity::class.java)
            if (bundle != null) {
                intent.putExtras(bundle)
            }
            context.startActivity(intent)
        }
    }
}
